lib/tests/packages-from-directory/default.nix

```diff
@@ -0,0 +1,43 @@
+{ lib ? import ../../.
+, pkgs ? import ../../../. {}
+}:
+let
+  inherit (pkgs) stdenv callPackage writeTextFile;
+  src = ./test-data;
+  packages = lib.packagesFromDirectory {
+    inherit callPackage;
+    directory = src;
+  };
```

Comment:
No need to involve derivations. The functions works for any Nix values. This should work:

```nix
packagesFromDirectory {
  packageFromFile = import;
```

And then just put strings directly into the package files.